pkg/storage/watch_cache.go

```diff
@@ -35,9 +35,10 @@ import (
 )
 
 const (
-	// MaximumListWait determines how long we're willing to wait for a
-	// list if a client specified a resource version in the future.
-	MaximumListWait = 60 * time.Second
+	// blockTimeout determines how long we're willing to block the request
+	// to wait for a given resource version to be propagated to cache,
+	// before terminating request and returning BadRequest error.
```

Comment:
If we do a short timeout, we need to return 429 server timeout with a retry after.  That's the best way to put back pressure on clients that need this feature and want to be somewhat naive.